### PR TITLE
refactor chunk reader

### PR DESCRIPTION
**Pull request by qiaojialin**
Remove hasNext() and only keep one getData in PageReader.
Remove unused parameter in BatchData

### CODE REVIEW

File: tsfile/src/test/java/org/apache/iotdb/tsfile/read/reader/PageReaderTest.java [hunk 1]
```diff
@@ -171,14 +171,11 @@ public void test(TSDataType dataType) {
...
+            new DeltaBinaryDecoder.LongDeltaDecoder(), null);
 
         int index = 0;
         long startTimestamp = System.currentTimeMillis();
-        BatchData data = null;
-        if (pageReader.hasNextBatch()) {
-          data = pageReader.nextBatch();
-        }
+        BatchData data = pageReader.getAllSatisfiedPageData();
         assert data != null;
```
Comment (jt2594838) [inline tsfile/src/test/java/org/apache/iotdb/tsfile/read/reader/PageReaderTest.java]: Replace with JUnit assertion.

File: server/src/main/java/org/apache/iotdb/db/query/reader/resourceRelated/NewUnseqResourceMergeReader.java [hunk 2]
```diff
@@ -88,16 +88,18 @@ public NewUnseqResourceMergeReader(Path seriesPath, TSDataType dataType,
...
-          QueryUtils.modifyChunkMetaData(currentChunkMetaDataList, pathModifications);
-        }
       } else {
-        // metadata list of already flushed chunk groups
+        // metadata list of already flushed chunks
         currentChunkMetaDataList = tsFileResource.getChunkMetaDataList();
       }
 
+      // get modifications and apply to chunk metadatas
+      List<Modification> pathModifications = context
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/query/reader/resourceRelated/NewUnseqResourceMergeReader.java]: need a ut? 
Or, in logical, can the PathModifications file be read before the data file is sealed?

File: tsfile/src/test/java/org/apache/iotdb/tsfile/read/reader/PageReaderTest.java (hunk 1 above)
Comment (qiaojialin) [reply tsfile/src/test/java/org/apache/iotdb/tsfile/read/reader/PageReaderTest.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/query/reader/resourceRelated/NewUnseqResourceMergeReader.java (hunk 2 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/query/reader/resourceRelated/NewUnseqResourceMergeReader.java]: no bug, I have added some test and comment

**Pull request merged**